File: apps/backend/app/schemas/explainability.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ExplainabilityRecordRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workflow_id: UUID
    agent_execution_id: UUID | None = None
    title: str
    summary: str
    reasoning_steps: list[str] = Field(default_factory=list)
    evidence: list[dict[str, Any]] = Field(default_factory=list)
    assumptions: list[str] = Field(default_factory=list)
    risks: list[str] = Field(default_factory=list)
    confidence_score: float
    created_at: datetime

File: apps/frontend/src/types/explainability.ts
export interface ExplainabilityRecord {
  id: string;
  workflow_id: string;
  agent_execution_id?: string | null;
  title: string;
  summary: string;
  reasoning_steps: string[];
  evidence: Array<Record<string, unknown>>;
  assumptions: string[];
  risks: string[];
  confidence_score: number;
  created_at: string;
}
